src/cpp/ext/otel/otel_plugin.h

```diff
@@ -217,53 +186,237 @@ class OpenTelemetryPluginBuilderImpl {
+  void AddCounter(
+      grpc_core::GlobalInstrumentsRegistry::GlobalUInt64CounterHandle handle,
+      uint64_t value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override;
+  void AddCounter(
+      grpc_core::GlobalInstrumentsRegistry::GlobalDoubleCounterHandle handle,
+      double value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override;
+  void RecordHistogram(
+      grpc_core::GlobalInstrumentsRegistry::GlobalUInt64HistogramHandle handle,
+      uint64_t value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override;
+  void RecordHistogram(
+      grpc_core::GlobalInstrumentsRegistry::GlobalDoubleHistogramHandle handle,
+      double value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override;
+  void SetGauge(
+      grpc_core::GlobalInstrumentsRegistry::GlobalInt64GaugeHandle handle,
+      int64_t value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override {}
+  void SetGauge(
+      grpc_core::GlobalInstrumentsRegistry::GlobalDoubleGaugeHandle handle,
+      double value, absl::Span<const absl::string_view> label_values,
+      absl::Span<const absl::string_view> optional_values) override {}
+  // TODO(yashkt, yijiem): implement async instrument.
+  void AddCallback(grpc_core::RegisteredMetricCallback* callback) override {}
+  void RemoveCallback(grpc_core::RegisteredMetricCallback* callback) override {}
+  grpc_core::ClientCallTracer* GetClientCallTracer(
+      const grpc_core::Slice& path, bool registered_method,
+      std::shared_ptr<grpc_core::StatsPlugin::ScopeConfig> scope_config)
+      override;
+  grpc_core::ServerCallTracer* GetServerCallTracer(
+      std::shared_ptr<grpc_core::StatsPlugin::ScopeConfig> scope_config)
+      override;
+
+  const absl::AnyInvocable<bool(const grpc_core::ChannelArgs& /*args*/) const>&
+  server_selector() const {
+    return server_selector_;
+  }
+  const absl::AnyInvocable<bool(absl::string_view /*target*/) const>&
+  target_attribute_filter() const {
+    return target_attribute_filter_;
+  }
+  const absl::AnyInvocable<bool(absl::string_view /*generic_method*/) const>&
+  generic_method_attribute_filter() const {
+    return generic_method_attribute_filter_;
   }
+  const std::vector<std::unique_ptr<InternalOpenTelemetryPluginOption>>&
+  plugin_options() const {
+    return plugin_options_;
+  }
+  Client& client() { return client_; }
+  Server& server() { return server_; }
 
-  std::bitset<64> active_mask_;
+ private:
+  // Instruments for per-call metrics.
+  Client client_;
+  Server server_;
+  // Instruments for non-per-call metrics.
+  struct Disabled {};
+  using InstrumentType = absl::variant<
```

Comment:
`InstrumentType` made me think that this is going to be an enum, but this is the actual OTel Instrument. Let's just call it `Instrument`?

Reply:
Done.